### PR TITLE
SDK-binding support for Tables worker extension 

### LINKED ISSUE #1767: Release Tables worker extension with SDK-binding support
- [ ] Merge Tables worker extension sdk-binding feature in to main
 - [ ] Release stable Tables worker extension package

### PR DESCRIPTION
**Pull request by liliankasem**
### Issue describing the changes in this PR

resolves #1767 

### Pull request checklist

* [x] My changes **do not** require documentation changes
  * [ ] Otherwise: Documentation issue linked to PR
* [x] My changes **should not** be added to the release notes for the next release
  * [ ] Otherwise: I've added my notes to `release_notes.md`
* [x] My changes **do not** need to be backported to a previous version
  * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [x] I have added all required tests (Unit tests, E2E tests)

### Additional information

Additional PR information

### CODE REVIEW

File: extensions/Worker.Extensions.Tables/src/Properties/AssemblyInfo.cs
Comment (liliankasem) [inline extensions/Worker.Extensions.Tables/src/Properties/AssemblyInfo.cs:7]: We're blocked on this WebJobs release

**Review by aishwaryabh — APPROVED** [review]
LGTM!

File: extensions/Worker.Extensions.Tables/src/Config/TablesBindingOptions.cs
Comment (mathewc) [inline extensions/Worker.Extensions.Tables/src/Config/TablesBindingOptions.cs:26]: Incorrect values being passed to the ANE ctor - should be a param name not an error message

File: extensions/Worker.Extensions.Tables/src/Config/TablesBindingOptionsSetup.cs
Comment (mathewc) [inline extensions/Worker.Extensions.Tables/src/Config/TablesBindingOptionsSetup.cs:39]: remove this unnecessary string concat

File: extensions/Worker.Extensions.Tables/src/TypeConverters/TableEntityEnumerableConverter.cs
Comment (mathewc) [inline extensions/Worker.Extensions.Tables/src/TypeConverters/TableEntityEnumerableConverter.cs:90]: Why are we materializing this ourselves instead of just letting the user enumerate themselves? The In-proc extension did this (looks like this code is a copy) because it was typed as JArray. But here the user is binding to an IEnumerable so we could just hand that back to them?

Comment (liliankasem) [reply extensions/Worker.Extensions.Tables/src/TypeConverters/TableEntityEnumerableConverter.cs:90]: Just to clarify, you're saying that the code was copied from [CreateJArray](https://github.com/Azure/azure-sdk-for-net/blob/f738f6a6b3f7f95114cc92fa26ded100907b74de/sdk/tables/Microsoft.Azure.WebJobs.Extensions.Tables/src/TablesExtensionConfigProvider.cs#L135), and in that code they did this because they had to create a JArray. We do not have to do this because `entities` is already IEnumerable so we should be able to just return that?

I agree with you here; I missed this in the original [PR](https://github.com/Azure/azure-functions-dotnet-worker/pull/1506), but returning entities directly here makes the most sense - thanks for catching!

Comment (liliankasem) [reply extensions/Worker.Extensions.Tables/src/TypeConverters/TableEntityEnumerableConverter.cs:90]: Though I wonder if we would still need to convert this to IEnumerable since technically `entities` is actually `ConfiguredCancelableAsyncEnumerable<TableEntity>`. Or if we should change the supported binding type to IAsyncEnumerable

Comment (liliankasem) [reply extensions/Worker.Extensions.Tables/src/TypeConverters/TableEntityEnumerableConverter.cs]: @mathewc thoughts on this ^ ? I updated the code but still creating an IList

Comment (mathewc) [reply extensions/Worker.Extensions.Tables/src/TypeConverters/TableEntityEnumerableConverter.cs]: You're pointing out that the entities is actually an `AsyncPageable<T>`? Then perhaps that's what we should be returning - there are methods on there related to paging that the user might care about. When we do a ToList, it ends up enumerating all the pages (possibly multiple service calls), right? Our goal with SDK bindings is to allow the user to bind to SDK natural types. So we need to decide if we want to interject ourselves and force enumerate/convert.

Comment (liliankasem) [reply extensions/Worker.Extensions.Tables/src/TypeConverters/TableEntityEnumerableConverter.cs]: Discussed offline, continuing with IEnumerable for now as we need to take into account Take for parity. Future work for IAsyncPageable

**Review by mattchenderson — APPROVED** [review]
Approving as these changes were previously reviewed going into the branch, and we've discussed the `IEnumerable` hydration offline. I have not gone through and conducted another review of the changes themselves.

**Pull request merged**
